src/main/java/persistence/entity/EntityPersisterImpl.java

```diff
@@ -16,25 +20,29 @@
 public class EntityPersisterImpl implements EntityPersister {
     private final GeneratedIdObtainStrategy generatedIdObtainStrategy;
     private final JdbcTemplate jdbcTemplate;
+    private final PersistentClass persistentClass;
 
-    public EntityPersisterImpl(GeneratedIdObtainStrategy generatedIdObtainStrategy, JdbcTemplate jdbcTemplate) {
+    public EntityPersisterImpl(
+            GeneratedIdObtainStrategy generatedIdObtainStrategy,
+            JdbcTemplate jdbcTemplate,
+            PersistentClass persistentClass
+    ) {
         this.generatedIdObtainStrategy = generatedIdObtainStrategy;
         this.jdbcTemplate = jdbcTemplate;
+        this.persistentClass = persistentClass;
     }
 
     @Override
     public boolean update(Object entity) {
-        TableData table = TableData.from(entity.getClass());
-        Columns columns = Columns.createColumnsWithValue(entity);
-        ColumnData keyColumn = columns.getPkColumn();
+        ColumnData keyColumn = persistentClass.getPkColumn();
 
-        if(keyColumn.getValue() == null) {
+        if(keyColumn.getValue(entity) == null) {
```

Comment:
```suggestion
        if(keyColumn.contains(entity)) {
```

음... 객체지향에서 값을 꺼내는것보다는 메시지를 보내는게 중요하니 이렇게 해보는것도 좋겠네요 ㅎ

Reply:
좋습니다!